packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx

```diff
@@ -232,36 +246,50 @@ export class DetailsHeaderBase extends BaseComponent<IDetailsHeaderBaseProps, ID
                 onClick={!isCheckboxHidden ? this._onSelectAllClicked : undefined}
                 aria-colindex={1}
                 role={'columnheader'}
-                aria-hidden={isCheckboxHidden ? true : undefined}
               >
                 {onRenderColumnHeaderTooltip(
                   {
                     hostClassName: css(classNames.checkTooltip),
                     id: `${this._id}-checkTooltip`,
                     setAriaDescribedBy: false,
                     content: ariaLabelForSelectAllCheckbox,
-                    children: (
+                    children: !isCheckboxHidden ? (
                       <DetailsRowCheck

```

Comment:
We're thinking it might be better to keep the styling within the `DetailsRowCheck` component and simply conditionally render its contents with either `isVisible` (currently sets style to `opacity: 1`) or introduce a new prop. Ideally, we could use `isVisible`. 

We need to confirm that this fix is correct from an accessibility POV. Once we do that, we can try to use `isVisible` and see if it regresses any public API / behavior (hopefully not).